weave/trace/serialization/custom_objs.py

```diff
@@ -47,25 +48,61 @@ def encode_custom_obj(obj: Any) -> dict | None:
         # Ensure load_instance is an op
         if not isinstance(serializer.load, Op):
             serializer.load = op(serializer.load)
-        # Save the load_intance_op
+        # Save the load_instance_op
         wc = require_weave_client()
 
         # TODO(PR): this can fail right? Or does it return None?
         # Calculating this URL is blocking, but we only have to pay it once per custom type
         load_instance_op_ref = wc._save_op(serializer.load, "load_" + serializer.id())  # type: ignore
         load_op_uri = load_instance_op_ref.uri()
 
-    encoded_path_contents = {
-        k: (v.encode("utf-8") if isinstance(v, str) else v)  # type: ignore
-        for k, v in art.path_contents.items()
-    }
-    return {
+    encoded = {
```

Comment:
The lack of typing here is unfortunate and makes it harder to understand what's going on.

Do I understand `encoded["val"]` to mean the inline-serialized object?

Reply:
Yes. I agree that stronger typing would be better but would rather leave that for follow up - I think there may need to be several changes because we do things like pop the `_type` key off the dict.

Reply:
Ok I wont block here, but I think it might be easier than you're describing (at least for MVP)

For poppable keys, you can write something like:
```py
class EncodedDict(TypedDict):
    _type: NotRequired[str]
    ...
```

Reply:
`NotRequired` is actually giving me issues in 3.9, I think there's some bad interaction with our code serialization.

Reply:
oh?  It was introduced in py311 so you might have to do import from `typing_extensions`

https://peps.python.org/pep-0655/

Reply:
if it still breaks then it's fine, I won't block here

Reply:
ok, unfortunate but I wont block.  Thanks for checking Jamie!